awx/main/credential_plugins/aim.py

```diff
@@ -98,7 +105,16 @@ def aim_backend(**kwargs):
             allow_redirects=False,
         )
     raise_for_status(res)
-    return res.json()['Content']
+    # CCP returns the property name capitalized, username is camel case
+    # so we need to handle that case
+    if object_property == '':
+        object_property = 'Content'
+    elif object_property.lower() == 'username':
+        object_property = 'UserName'
+    else:
+        object_property = object_property.capitalize()
+
+    return res.json()[object_property]
```

Comment:
should we check that `object_property` is a key in the response json object, and maybe raise a 400 error if it isn't. otherwise we'll get a 500 server error if a user supplies a bogus `object_property` value

Reply:
Is this necessary if on L83 the key has a defined default of `''`?

`object_property = kwargs.get('object_property', '')`

In the conditional block you highlighted, the logic goes:

1. If `object_property` is `''` then set it to `Content`.
2. If the `object_property` is `username` then set it to `UserName`.
3. Otherwise, we can assume a value is given other than nothing or `username`, so we just need to capitalize the first letter of the value.

I don't believe, since there's a default value given on L83, that there will ever be an instance where `object_property` is not a key once we evaluate the highlighted conditional you are referring to.

Please correct me if my thinking is wrong here.  Thanks for the feedback!

Reply:
I could be wrong, so I apologize in advance if I misunderstand something

```python3
object_property = kwargs.get('object_property', '')
```

this handles just the _input_ side. that is, if the input from user _doesn't_ provide `object_property`

but what if a user puts `bogus_value` on the object property prompt in the UI?

seems this like will hit a syntax error

`res.json()[object_property]`

because the response from cyberark isn't going to have a "bogus_value" key

I don't have a cyberark server to try this on, but curious what happens if you provide a nonsense value in the UI for object_property

Reply:
Ahhhhh, excellent point!  I see what you mean now.  You were referring to the return of the `object_property` in the response rather than the conditional logic.

Let me test what the outcome of this occurring would be and we can continue the discussion here.

Reply:
The error that occurs when a key not contained in the JSON response is used as the value for `object_property` is:
![image](https://user-images.githubusercontent.com/1924063/223493927-66f13f07-c0fc-422d-9536-be5dc1f676b6.png)

@fosterseth Any recommendations for handling this if it is not an acceptable exception?

Reply:
@infamousjoeg thanks for the screenshots that is helpful

the "test" button will call out to the /test/ endpoint, and that specific code is wrapped in a `try except` block, so the backend is still probably hitting a 500 syntax error type of problem but it doesn't show up there.

instead of using the Test, what happens if you try it out in a job run? that is, just save the form, then create a machine credential with this lookup and provide the `thisdoesnotexist` as the object property and save it. Try to run a job and see what happens (and note if any server errors pop up)

Reply:
It's been a rough few weeks. I'm continuing this today...

Reply:
I think we can either raise a 400 Bad Request if 'object_property' is not in `res.json()`, or we can look for key 'Content' (as we do in empty case)

Reply:
`ValueError` is raised when an Object Property is provided that is not returned in the JSON response.
![image](https://user-images.githubusercontent.com/1924063/231207612-ee43ed7a-77f8-4a83-8e73-978dafac82b0.png)

Reply:
Maybe a KeyError would be preferred?  Could not figure out how to raise a 400 Bad Request without introducing additional dependencies.

Reply:
yeah let's do a `KeyError` instead. Don't worry about raising a Bad Request for this

Reply:
Committed the change to KeyError to PR.